### PR TITLE
[Docs] Add docs to deploy project on Astro Cloud

### PR DESCRIPTION
**Pull request by pankajastro**
Add documentation for deploying the project in the /dev directory on Astro Cloud. I created a Makefile target called "deploy" for this purpose.

I set up an Astro deployment on AWS Cloud and a RayCluster on GKE for this exercise.

To deploy on Astro Cloud, run the command make deploy.

**Generated Docs**

Screenshot 2024-11-01 at 12 55 21 AM

closes: https://github.com/astronomer/astro-provider-ray/issues/88

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,6 +22,14 @@ docker-run: build-whl ## Runs local Airflow for testing
 		cd dev && astro dev restart; \
 	fi
 
+.PHONY: astro-login
+astro-login: # Login to Astro cloud
+	cd dev && astro login
```
Comment (pankajkoti) [inline Makefile:27]:
```suggestion
	cd dev && astro login cloud.astronomer-stage.io
```
Probably better to login to stage for dev deployments?

File: dev/Dockerfile.ray_google_cloud [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        apt-get install python3 google-cloud-sdk -y && \
+        apt-get install google-cloud-sdk-gke-gcloud-auth-plugin -y
+
+RUN curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 && \
+    chmod 700 get_helm.sh && \
+    ./get_helm.sh
+
+USER astro
+
+RUN pip install /usr/local/airflow/include/*.whl
```
Comment (pankajkoti) [inline dev/Dockerfile.ray_google_cloud:22]:
```suggestion
RUN pip install /usr/local/airflow/include/*.whl

ENV AIRFLOW__CORE__TEST_CONNECTION=Enabled
```
Let's also add this for dev help, WDYT?

File: docs/getting_started/astro_setup.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+Steps
+-----
+
+1. **Create a Google Cloud service account key JSON file:** Follow the instructions `here <https://cloud.google.com/iam/docs/keys-create-delete>`_
+
+2. **Set up the gcloud credentials**
+
+.. code-block:: bash
+
+    export GOOGLE_APPLICATION_CREDENTIALS=/path/to/your/service-account-key.json
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/getting_started/astro_setup.rst]: Removed it

File: docs/getting_started/astro_setup.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+4. **Create a GKE cluster**
+
+.. code-block:: bash
+
+    gcloud container clusters create cluster-name \
+          --zone us-central1-a \
+          --num-nodes 1 \
+          --machine-type e2-standard-4 \
+          --enable-autoscaling --min-nodes=1 --max-nodes=3 \
+          --no-enable-ip-alias
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/getting_started/astro_setup.rst:53]: Added this command

File: dev/Dockerfile.ray_google_cloud
[the remark this answers is not in the document]
Comment (pankajastro) [reply dev/Dockerfile.ray_google_cloud:1]: As discussed offline  moved content from this file to Dockerfile

File: docs/getting_started/astro_setup.rst [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+Steps
+-----
+
+1. **Create an Astro deployment:** Follow the instructions `here <https://www.astronomer.io/docs/astro/create-deployment#:~:text=Create%20a%20Deployment%E2%80%8B,%2C%20executor%2C%20and%20worker%20resources.>`_
+
+2. **Deploy the project on Astro Cloud**
+
+
+Please ensure you use `Dockerfile.ray_google_cloud <https://github.com/astronomer/astro-provider-ray/blob/main/dev/Dockerfile.ray_google_cloud>`_ as the default. The command below uses `Dockerfile <https://github.com/astronomer/astro-provider-ray/blob/main/dev/Dockerfile>`_ by default.
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/getting_started/astro_setup.rst]: As discussed offline moved content of Dockerfile.ray_google_cloud file to the Dockerfile

File: docs/getting_started/astro_setup.rst [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    gcloud auth activate-service-account --key-file=/path/to/your/service-account-key.json
+
+3. **Install the gcloud auth plugin**
+
+.. code-block:: bash
+
+    gcloud components install gke-gcloud-auth-plugin
+
+4. **Create a GKE cluster**
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/getting_started/astro_setup.rst:44]: Added a command

File: docs/getting_started/astro_setup.rst [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+.. code-block:: bash
+
+    make deploy
+
+This command will build a wheel from your branch and deploy the `project <https://github.com/astronomer/astro-provider-ray/tree/main/dev>`_ in Astro Cloud
+
+3. **Create an Airflow Connection**
+
+- Navigate to Admin -> Connections -> Add a new record. Select the connection type ``Ray`` and set the parameter ``Kube config path`` to the path of ``kubeconfig.yaml``.
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/getting_started/astro_setup.rst:101]: No, we do not need

File: dev/Dockerfile.ray_google_cloud [hunk 8]
```diff
@@ -0,0 +1,22 @@
+FROM quay.io/astronomer/astro-runtime:12.2.0
+
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply dev/Dockerfile.ray_google_cloud]: As discussed offline moved the content of Dockerfile.ray_google_cloud file to the Dockerfile and deleted this file

File: docs/getting_started/astro_setup.rst [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+5. **Retrieve GKE cluster configuration**
+
+.. code-block:: bash
+
+    gcloud container clusters get-credentials cluster-name --zone us-central1-a
+
+    kubectl config view --raw > kubeconfig.yaml
+
+We will use this ``kubeconfig.yaml`` to create an Airflow connection type ``Ray`` in Astro Cloud.
```
[the remark this answers is not in the document]
Comment (pankajastro) [reply docs/getting_started/astro_setup.rst]: Added some in commit 45d847e5bba6e0008e2da413f5582dd2b37a8fed

**Review by pankajkoti — APPROVED** [review]
LGTM. Thanks for addressing all the review comments!

**Pull request merged**